### PR TITLE
feat: rpm weak dependencies

### LINKED ISSUE #1333: rpm: add info about package weak dependencies
Let's add info about rpm-package weak (or soft) dependencies. 
Tags names and values can be found [here](https://rpm-software-management.github.io/rpm/manual/tags.html).

Check some other docs also:
https://docs.fedoraproject.org/en-US/packaging-guidelines/WeakDependencies/
https://rpm-software-management.github.io/rpm/manual/dependencies.html

### PR DESCRIPTION
**Pull request by olenagerasimova**
closes #1333 

Added info about rpm package weak dependencies (recommends, suggests, supplements and enhances tags) into rpm index primary.xml. 

Also, simplified some classes/methods by reducing code duplicates.

Corrected conda-adapter tests: there are some conflicts between conda and anaconda settings files (we use both). These conflicts are resolved automatically with warnings and do not affect the tests.

### CODE REVIEW

File: conda-adapter/src/test/java/com/artipie/conda/CondaSliceITCase.java [hunk 1]
```diff
@@ -120,7 +119,7 @@ void anacondaCanLogin() throws Exception {
             new StringContainsInOrder(
                 new ListOf<>(
                     "Using Anaconda API: http://host.testcontainers.internal",
-                    "any's login successful"
+                    "any's", "login successful"
```
Comment (dgarus) [inline conda-adapter/src/test/java/com/artipie/conda/CondaSliceITCase.java:122]: Is these changes related to #1336?

**Review by dgarus — APPROVED** [review]

File: conda-adapter/src/test/java/com/artipie/conda/CondaSliceITCase.java (hunk 1 above)
Comment (olenagerasimova) [reply conda-adapter/src/test/java/com/artipie/conda/CondaSliceITCase.java:122]: yes, I've corrected conda tests to pass

**Review by dgarus — APPROVED** [review]

**Comment by olenagerasimova** [comment]
@dgarus please, approve

**Review by dgarus — APPROVED** [review]
